### PR TITLE
Fix/scansion

### PR DESCRIPTION
**Pull request by alvp**
Added syllabification exceptions for unstressed words
Addep enable/disable suppport for spacy_affices
Added check to see if las stressed word belongs to last word
Fixed tests

### CODE REVIEW

File: src/rantanplan/core.py [hunk 1]
```diff
@@ -78,11 +78,11 @@
                                        'mas', 'me', 'mi', 'nos', 'os', 'que',
                                        'se', 'si',
                                        'su', 'tan', 'te', 'tu', "tus", "oh"}
-UNSTRESSED_FORMS = {"que", "cual", "quien", "donde", "cuando", "cuanto",
```
Comment (versae) [inline src/rantanplan/core.py]: So all of those go out?

File: src/rantanplan/core.py [hunk 2]
```diff
@@ -500,32 +516,41 @@ def get_word_stress(word, pos, tag, alternative_syllabification=False):
                      or (pos in ("PRON", "DET")
                          and tag.get("PronType", None) in (
                                  "Exc", "Int", "Dem"))
-                     or "".join(word).lower() in POSSESSIVE_PRON)):
+                     or "".join(
+                            word).lower() in STRESSED_PRON) and word_lower not in UNSTRESSED_FORMS):
```
Comment (versae) [inline src/rantanplan/core.py]: That's a weird way of breaking the line, isnt?

File: src/rantanplan/core.py (hunk 1 above)
Comment (alvp) [reply src/rantanplan/core.py]: Yep

File: src/rantanplan/core.py (hunk 2 above)
Comment (alvp) [reply src/rantanplan/core.py]: Woopsie, will fix

File: src/rantanplan/core.py [hunk 3]
```diff
@@ -583,18 +623,55 @@ def get_words(word_list, alternative_syllabification=False):
...
     return syllabified_words
 
 
+def join_affixes(line):
+    """
+    TODO
+    :param line:
+    :return:
+    """
+    new_syllabified_words = []
```
Comment (versae) [inline src/rantanplan/core.py]: Maybe the `new_` is not needed anymore?

File: src/rantanplan/core.py [hunk 4]
```diff
@@ -583,18 +623,55 @@ def get_words(word_list, alternative_syllabification=False):
...
+            new_syllabified_words.append(word)
+        else:
+            indices_to_ignore = range(index, index + affixes_length+1)
+            join_word = []
+            for affix_index in indices_to_ignore:
+                affix = line[affix_index]['word']
+                join_word += [syll["syllable"] for syll in affix]
+            new_stress = get_word_stress("".join(join_word), word["pos"],
+                                         word["tag"])
+            new_stress["word"][-1]["is_word_end"] = True
```
Comment (versae) [inline src/rantanplan/core.py]: Rename `new_stress` as `word_stress`?

File: src/rantanplan/core.py [hunk 5]
```diff
@@ -583,18 +623,55 @@ def get_words(word_list, alternative_syllabification=False):
...
+    new_syllabified_words = []
+    indices_to_ignore = []
+    for index, word in enumerate(line):
+        affixes_length = word.get('affixes_length', None)
+        if index in indices_to_ignore:
+            continue
+        elif affixes_length is None:
+            new_syllabified_words.append(word)
+        else:
+            indices_to_ignore = range(index, index + affixes_length+1)
```
Comment (versae) [inline src/rantanplan/core.py]: Missing spaces in `range(index, index + affixes_length + 1)`

File: src/rantanplan/core.py [hunk 6]
```diff
@@ -710,7 +788,7 @@ def generate_phonological_groups(tokens):
                         yield groups
                     else:
                         for liaison_positions_2 in generate_liaison_positions(
-                            syllables, liaison[1]
+                                syllables, liaison[1]
```
Comment (versae) [inline src/rantanplan/core.py]: Is this OK? Two indentations?

File: src/rantanplan/core.py [hunk 7]
```diff
@@ -692,9 +770,9 @@ def generate_phonological_groups(tokens):
         syllables = get_syllables_word_end(words)
         for liaison in (
                 ("synalepha",),
+                ("synalepha", "sinaeresis"),
                 ("sinaeresis",),
                 ("sinaeresis", "synalepha"),
-                ("synalepha", "sinaeresis"),
```
Comment (versae) [inline src/rantanplan/core.py]: Please, document why the change.

File: src/rantanplan/pipeline.py [hunk 8]
```diff
@@ -29,21 +29,25 @@ def custom_tokenizer(nlp):
 _load_pipeline = {}
 
 
-def load_pipeline(lang=None):
+def load_pipeline(lang=None, use_affixes=True):
     """
     Loads the new pipeline with the custom tokenizer
     :param lang: Spacy language model
+    :param use_affixes: Wether or not use spacy_affixes
```
Comment (versae) [inline src/rantanplan/pipeline.py]: ` Whether or not affixes should be treated separately.`

File: src/rantanplan/pipeline.py [hunk 9]
```diff
@@ -29,21 +29,25 @@ def custom_tokenizer(nlp):
...
-        suffixes = {k: v for k, v in load_affixes().items() if k.startswith(AFFIXES_SUFFIX)}
-        affixes_matcher = AffixesMatcher(nlp, split_on=["VERB"], rules=suffixes)
-        nlp.add_pipe(affixes_matcher, name="affixes", first=True)
+        if use_affixes:
+            nlp.tokenizer = custom_tokenizer(nlp)
+            nlp.remove_pipe("affixes") if nlp.has_pipe("affixes") else None
+            suffixes = {k: v for k, v in load_affixes().items() if
+                        k.startswith(AFFIXES_SUFFIX)}
+            affixes_matcher = AffixesMatcher(nlp, split_on=["VERB"],
+                                             rules=suffixes)
```
Comment (versae) [inline src/rantanplan/pipeline.py]: The line `nlp.tokenizer = custom_tokenizer(nlp)` should remain out of the `if` block I believe.

File: tests/test_pipeline.py [hunk 10]
```diff
@@ -13,15 +13,15 @@
...
         nlp.vocab.lookups.get_table = lambda *_: {}
         return nlp
 
     monkeypatch.setattr(spacy, 'load', mockreturn)
     # lang doesn't matter as long as it hasn't been used in the test session
-    nlp = load_pipeline("blank")
+    nlp = load_pipeline("blank", use_affixes=True)
     doc = nlp("prue-\nba")
     token_dict = []
     for token in doc:
```
Comment (versae) [inline tests/test_pipeline.py]: Test for the opposite case is missing.

File: src/rantanplan/core.py [hunk 11]
```diff
@@ -292,7 +295,8 @@ def get_stresses(phonological_groups):
     # Get position for the last syllable of the penultimate word
     if last_word_syllables.count(True) > 1:
         penultimate_word = -(
-            [i for i, n in enumerate(last_word_syllables[::-1]) if n][1] + 1)
+                [i for i, n in enumerate(last_word_syllables[::-1]) if n][
+                    1] + 1)
```
Comment (versae) [inline src/rantanplan/core.py]: Please, put back to its original form.

File: src/rantanplan/core.py [hunk 12]
```diff
@@ -647,29 +649,29 @@ def get_words(word_list, alternative_syllabification=False):
 
 def join_affixes(line):
     """
-    TODO
-    :param line:
-    :return:
+    Join affixes of split words and recalculates stress
+    :param line: List of syllabified words (dict)
+    :return: Dictionary with a list of tokens with joined affixes
```
Comment (versae) [inline src/rantanplan/core.py]: Returns a list as well.

File: tests/test_pipeline.py [hunk 13]
```diff
@@ -13,15 +13,15 @@
...
         nlp.vocab.lookups.get_table = lambda *_: {}
         return nlp
 
     monkeypatch.setattr(spacy, 'load', mockreturn)
     # lang doesn't matter as long as it hasn't been used in the test session
-    nlp = load_pipeline("blank", use_affixes=True)
+    nlp = load_pipeline("blank")
     doc = nlp("prue-\nba")
     token_dict = []
     for token in doc:
```
Comment (versae) [inline tests/test_pipeline.py]: Please, add the opposite test.
